demo/src/store/store.js

```diff
@@ -0,0 +1,95 @@
+import {createStore} from 'redux';
+import airplaneEars from '../images/airplane_ears.jpg';
+import bigYawn from '../images/big_yawn.jpg';
+import fluffy from '../images/fluffy.jpg';
+import makinBiscuits from '../images/makin_biscuits.jpg';
+import pawsUp from '../images/paws_up.jpg';
+import petMyBelly from '../images/pet_my_belly.jpg';
+import phoneBackground from '../images/phone_background.jpg';
+import snoozing from '../images/snoozing.jpg';
+import spoiled from '../images/spoiled.jpg';
+import whatYouLookinAt from '../images/what_you_lookin_at.jpg';
+
+/**
+ * Functions to update the value saved in the store.
+ * @param {!Object} state The state to take in initially.
+ * @param {{type:string, id:string, amount:?number}} action a description of
+ * the action to take.
+ * @return {!Object} the new State
+ */
+function reducer(state = [], action) {
+  // deep copy state so as not to mutate anything
+  const newState = JSON.parse(JSON.stringify(state));
```

Comment:
things like this usually belong to utils where they could be properly tested and well documented. (deep copy using JSON.parse/stringify has its own edge-cases)

Reply:
I moved this to a different file, I'm still using JSON.parse/stringify since it works for now